### DISCUSSION TITLE
Output-print heavily affected if CPU is stressed, why?

### DISCUSSION THREAD

**Discussion by MrM40** [ipmitool/ipmitool#275] [Q&A]
I relation to the Issue thread [#260](https://github.com/ipmitool/ipmitool/issues/260) and the follwong answer from@AlexanderAmelkin:

> ipmitool sends a lot of requests over IPMI device to perform this "simple" task. That means that it executes a lot of syscalls, each of which is per se a point where the kernel's scheduler may switch to another process. Also, inside those syscalls there is a lot of waiting for response from the BMC, which (the waiting) is definitely the point where the control is switched to another process by the scheduler.
>
>Any program doing I/O operations should be affected more or less similarly in these circumstances."

So the good question is will calling the ipmitool repeatedly. e.g, for a custom made monitor/alarm service,  put a proportional high "stress"on system/other processes?
That's actually what I'm worried about. I wanna retrieve temperatures and FAN speeds, which are available through IPMI on my SuperMicro X11 boards

**Comment by AlexanderAmelkin** [comment]
I would be more worried about the burden that you'd put on your BMC by such calls, especially if you do it every short interval (like several times a minute). Sensors usually don't change that fast, and the more correct way to subscribe for significant changes (such as threshold crossing) would be to set up PEF is supported by your BMC.

On the client side repeatedly calling ipmitool must not be too CPU-stressful. It will mostly be waiting for responses from the BMC.

**Reply by MrM40** [reply]
OK Thanks.

Can you point me in a direction with PEF? Can see PEF in mentioned in several IPMI tool manual but I'm still not aware what that is?

By second thought I guess the optimal way of getting CPU temp and FAN speeds would be to probe the  IPMI/BMC directly from my custom monitor tool (.Net Core, C#), instead of using an external tool which start a new process every time. It's just for server monitoring, so 2 times a minute would be adequate.
It might be out of my coding abilities but it would optimal.

**Reply by AlexanderAmelkin** [reply]
You can read the publicly available IPMI 2.0 Specification regarding PEF. In short, those are platform event filters that can bind various actions to certain conditions.

As for calling IPMI from your program, that's definitely not about `ipmitool`. Take a look at `freeipmi` or `OpenIPMI`.
